_posts/2017-06-01-Generative-Adversial-Network-in-R.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: post
+title: "Conditional Generative Adversial Network with MXNet R package"
+date:   2017-06-01
+author: Jeremie Desgagne-Bouchard
+categories: rstats
+comments: true
+---
+
+This tutorial shows how to build and train a Conditional Generative Adversial Network (CGAN) on MNIST images. 
+
+### How GAN works
+
+A Generative Adversial Model simultaneously trains two models: a generator that learns to output fake samples from an unknown distribution and a discriminator that learns to distinguish fake from real samples. 
+
+The CGAN is a conditional variation of the GAN where the generator is instructed to generate a real sample having specific characteristics rather than a generic sample from full distribution. Such condition could be the label associated with an image like in this tutorial or a more detailed tag as shown in the example below: 
+
+![](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/blog_R_cgan/cgan_network.jpg)
+
+Image credit: [Scott Reed](https://github.com/reedscot/icml2016)
+
+### Initial setup
+
+The following packages are needed to run the tutorial: 
+
+```r
+require("imager")
+require("dplyr")
+require("readr")
+require("mxnet")
+```
+
+The full demo is comprised of the two following scripts available on [GitHub](https://github.com/jeremiedb/gan_example): 
```

Comment:
@jeremiedb Do you mind sending a PR to put the code into https://github.com/dmlc/mxnet/tree/master/example